### ISSUE TITLE
BUG: Workflow not overwriting the README

### ISSUE THREAD

**Issue by andrea-acampora** [anmol098/waka-readme-stats#292]
The workflow runs with success but it doesn't overwrite the README with newest info.
You can check this behaviour viewing the commit list  at this [repo](https://github.com/andrea-acampora/andrea-acampora)

**Comment by ArslanAmeer** [comment]
@andrea-acampora @anmol098 
Any update on this?
i am manually running wakatime workflow, but it is not overwriting the README.

Error in logs with the successful workflow :

```
Traceback (most recent call last):
  File "/main.py", line 548, in <module>
    username = user_data["data"]["viewer"]["login"]
KeyError: 'data'
Exception Occurred 'data'
```

**Comment by andrea-acampora** [comment]
@ArslanAmeer I think that it generates and overwrites the README only if there'are some changes.

**Comment by ArslanAmeer** [comment]
@andrea-acampora Waka dashboard show changes, but the README file doesn't .
I don't know for how long the stats hasn't been updated, because there is huge diff in what waka dashboard is showing

**Comment by andrea-acampora** [comment]
Try to manually delete the section between  and . Then launch the workflow and it will generate the waka section.

**Comment by ArslanAmeer** [comment]
@andrea-acampora  Same. not working. Same error as mentioned above

**Comment by andrea-acampora** [comment]
Your error refers to this issue #305. Btw in my repo the action works fine so i really don't know.

**Comment by ArslanAmeer** [comment]
Well, it worked after auto run.
Manually triggering the workflow does not update README.md file

**Comment by exilvm** [comment]
> @andrea-acampora @anmol098 Any update on this? i am manually running wakatime workflow, but it is not overwriting the README.
> 
> Error in logs with the successful workflow :
> 
> ```
> Traceback (most recent call last):
>   File "/main.py", line 548, in <module>
>     username = user_data["data"]["viewer"]["login"]
> KeyError: 'data'
> Exception Occurred 'data'
> ```

The query throws this error:

`{'errors': [{'type': 'INSUFFICIENT_SCOPES', 'locations': [{'line': 5, 'column': 7}], 'message': "Your token has not been granted the required scopes to execute this query. The 'email' field requires one of the following scopes: ['user:email', 'read:user'], but your token has only been granted the: ['repo', 'workflow', 'write:packages'] scopes. Please modify your token's scopes at: https://github.com/settings/tokens."}]}`

So basically granting your token the missing field (which in my case "email") will fix the issue.
I will implement error handeling on the query so the workflow logs will show feedback on what token scopes are missing.

**Comment by ArslanAmeer** [comment]
@exilvm but i have already granted it

**Comment by exilvm** [comment]
> @exilvm but i have already granted it

Well then there could be other scopes missing.

**Comment by ArslanAmeer** [comment]
@exilvm  There should be mentioned on Readme that which exactly scopes are required.

**Comment by exilvm** [comment]
Yep.
